library/Icingadb/Common/Macros.php

```diff
@@ -120,6 +121,12 @@ public function resolveMacro(string $macro, $object): string
             $value = null;
         }
 
-        return $value !== null ? $value : $macro;
+        if ($value instanceof DateTime) {
+            $value = $value->getTimestamp();
+        } elseif (is_bool($value)) {
+            $value = $value ? 'y' : 'n';
+        }
+
+        return $value ?? $macro;
```

Comment:
Cast to string here, it might still be an int or float I assume.

Reply:
The specified return type of the function is responsible for the cast.

Reply:
Hm, yeah, as if [implicit type conversion did never turn out to be an issue](https://php.watch/versions/8.1/deprecate-implicit-conversion-incompatible-float-string).

Explicit is better than implicit.